src/SymfonyCache/HeaderReplaySubscriber.php

```diff
@@ -108,8 +110,17 @@ public function preHandle(CacheEvent $event)
             $headersToReplay = explode(',', $preflightResponse->headers->get(HeaderReplayListener::REPLAY_HEADER_NAME));
 
             foreach ($headersToReplay as $header) {
+
+                // Make sure Set-Cookie is never considered because we do this
+                // manually
+                $header = strtolower($header);
+                unset($headersToReplay['set-cookie']);
```

Comment:
Should the `unset()` not be called before the loop?

Reply:
Ah, that was an issue I would've found with unit tests anyway :)